### PR TITLE
refactor: overhaul of format.sh

### PR DESCRIPTION
**Pull request by alexeagle**
Use GitHub Linguist to define the file extensions for each language, getting much more accurate. Extract a helper function for listing files we want to format to reduce duplication.

---

### Type of change

- Refactor (a code change that neither fixes a bug or adds a new feature)

### Test plan

- Manual testing; please provide instructions so we can reproduce:
Ran format.sh after changing code in each language

### CODE REVIEW

File: format/private/mirror_linguist_languages.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+# Generate a snippet of bash code to help us detect languages based on filenames
+raw=$(mktemp --suffix=.yml)
+json=$(mktemp --suffix=.json)
+wget -O "$raw" https://raw.githubusercontent.com/github-linguist/linguist/master/lib/linguist/languages.yml
+yq -o=json '.' "$raw" > "$json"
```
Comment (alexeagle) [inline format/private/mirror_linguist_languages.sh:6]: I'm sure YQ could do this in one step, but I'm already familiar with JQ and it wasn't the right day to learn a new tool.

Comment (thesayyn) [reply format/private/mirror_linguist_languages.sh]: we don't need yq anymore. we can continue using jq

Comment (alexeagle) [reply format/private/mirror_linguist_languages.sh]: the input is a yaml file though, so I think I do need this conversion?

Obviously this code will barely ever be run, and one could just look at the github-linguist repo and see a short list of extensions for the next language added. Maybe it's overkill to check this in at all?

File: format/private/format.sh [hunk 2]
```diff
@@ -19,32 +22,72 @@ function on_exit {
...
+      'Protocol Buffer') patterns=('*.proto') ;;
+      'Python') patterns=('.gclient' 'DEPS' 'SConscript' 'SConstruct' 'wscript' '*.py' '*.cgi' '*.fcgi' '*.gyp' '*.gypi' '*.lmi' '*.py3' '*.pyde' '*.pyi' '*.pyp' '*.pyt' '*.pyw' '*.rpy' '*.spec' '*.tac' '*.wsgi' '*.xpy') ;;
+      'SQL') patterns=('*.sql' '*.cql' '*.ddl' '*.inc' '*.mysql' '*.prc' '*.tab' '*.udf' '*.viw') ;;
+      'Scala') patterns=('*.scala' '*.kojo' '*.sbt' '*.sc') ;;
+      'Shell') patterns=('.bash_aliases' '.bash_functions' '.bash_history' '.bash_logout' '.bash_profile' '.bashrc' '.cshrc' '.flaskenv' '.kshrc' '.login' '.profile' '.zlogin' '.zlogout' '.zprofile' '.zshenv' '.zshrc' '9fs' 'PKGBUILD' 'bash_aliases' 'bash_logout' 'bash_profile' 'bashrc' 'cshrc' 'gradlew' 'kshrc' 'login' 'man' 'profile' 'zlogin' 'zlogout' 'zprofile' 'zshenv' 'zshrc' '*.sh' '*.bash' '*.bats' '*.cgi' '*.command' '*.fcgi' '*.ksh' '*.sh.in' '*.tmux' '*.tool' '*.trigger' '*.zsh' '*.zsh-theme') ;;
+      'Starlark') patterns=('BUCK' 'BUILD' 'BUILD.bazel' 'MODULE.bazel' 'Tiltfile' 'WORKSPACE' 'WORKSPACE.bazel' '*.bzl' '*.star') ;;
```
Comment (thesayyn) [inline format/private/format.sh:48]: hmm, BUCK files might contain types. does our formatter understand that?

Comment (alexeagle) [reply format/private/format.sh:48]: This is a problem generally, we use a tool that "formats code in some language" but there are syntactic variants for all of them that we can't possibly be experts on.
I think maybe the right answer (in a follow-up PR?) is to provide an affordance for this list of patterns to be overridden from the `multi_formatter_binary` rule

File: format/private/mirror_linguist_languages.sh (hunk 1 above)
Comment (thesayyn) [reply format/private/mirror_linguist_languages.sh]: oh missed that the extension was yaml.

**Review by thesayyn — APPROVED** [review]

**Pull request merged**